api4/post.go

```diff
@@ -120,16 +120,20 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	afterPost := r.URL.Query().Get("after")
-	beforePost := r.URL.Query().Get("before")
-	sinceString := r.URL.Query().Get("since")
+	if len(afterPost) > 0 && len(afterPost) != 26 {
+		c.SetInvalidParam("after")
+		return
+	}
 
-	channelId := c.Params.ChannelId
-	page := c.Params.Page
-	perPage := c.Params.PerPage
+	beforePost := r.URL.Query().Get("before")
+	if len(beforePost) > 0 && len(beforePost) != 26 {
+		c.SetInvalidParam("before")
+		return
+	}
```

Comment:
Added validation on these query parameters so that it returns an error when passing invalid parameters.  This also avoids returning invalid next/post IDs.